### PR TITLE
Fix the DatadogAgent CRD for clusters < 1.21

### PR DESCRIPTION
**Pull request by CharlyF**
#### What this PR does / why we need it:

We had not updated the CRD following up on the update to store v2alpha1 https://github.com/DataDog/helm-charts/pull/798

#### Which issue this PR fixes

  - fixes https://github.com/DataDog/datadog-operator/issues/649

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] Chart Version bumped
- [x] `CHANGELOG.md` has been updated
- [x] Variables are documented in the `README.md`

### CODE REVIEW

File: charts/datadog-crds/templates/datadoghq.com_datadogagents_v1beta1.yaml [hunk 1]
```diff
@@ -39,7 +39,6 @@ spec:
   scope: Namespaced
   subresources:
     status: {}
-  version: v1alpha1
```
Comment (CharlyF) [inline charts/datadog-crds/templates/datadoghq.com_datadogagents_v1beta1.yaml:42]: QA'ing that this is not an issue.

Comment (CharlyF) [reply charts/datadog-crds/templates/datadoghq.com_datadogagents_v1beta1.yaml:42]: Checked

**Review by khewonc — APPROVED** [review]

**Comment by CharlyF** [comment]
CI failure for 1.18.20 is strange but is not due to this change. We will investigate on the side.

**Comment by dlydiard** [comment]
This broke installations using kustomize --enable-helm
Even though we are on kube 1.23, it's trying to use the v1beta1 CRD instead of v1.
Please allow override of CRD version via values.yaml for the CRD charts.

**Pull request merged**